torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -94,19 +94,69 @@ struct NodeCalls : public std::unordered_map<Node*, NodeCall> {
   uint32_t _next_id = 0;
 };
 
-struct AutogradCompilerCall {
-  AutogradCompilerCall(bool accumulate_grad_)
-      : accumulate_grad(accumulate_grad_),
-        default_dyn_type(SizeInput::STATIC) {}
+struct TensorArg {
+  // Represents a de-duplicated tensor that will be passed into the graph
+  TensorArg(uint32_t i = 0) : id(i) {}
+  uint32_t index() const {
+    TORCH_INTERNAL_ASSERT(defined());
+    return id - 1;
+  }
+  bool defined() const {
+    return id != 0;
+  }
+  uint32_t id;
+  at::Tensor proxy_tensor;
+};
+
+struct TensorArgs {
+  // Manages a collection of TensorArgs and mappings from Tensors/SavedVariables
+  // to them
```

Comment:
Should we mention that the whole point is to be able to unpack() saved_var exactly one and keep a mapping to the unpacked Tensor?